### ISSUE TITLE
Is it possible to have ereader available when file sharing 

### ISSUE THREAD

**Issue by dnwk** [Yetangitu/owncloud-apps#76]
Is it possible to have e-reader available on the sharing page so that it will render the book in reading mode instead of just downloading the pdf

**Comment by Yetangitu** [comment]
It should already do so, if it doesn't it is a bug. Which platform (NC or OC, which version) do you use?

I just tested it, it works as intended for me. The following image shows a book (in EPUB format) shared through a link, opened in another browser which is not logged in to NC (it also works when logged in but for users who can log in you would not need to share by link).

![image](https://user-images.githubusercontent.com/2754838/35761795-2593f784-088d-11e8-8bd2-ecdc8165898b.png)

Try the following: share a document by link from your account, open another browser (not just another window, a browser which is not logged in to NC/OC to make sure you test under the same conditions as the file recipient would be) and open the share link in that browser. You should now get something resembling the above image. If you only get the _Add to your xxxcloud_ and _Download_ options something is not working the way I intended.

**Comment by dnwk** [comment]
I am using NC 12.0.4. It is working for EPUB but not for PDFs.

**Comment by Yetangitu** [comment]
Are you getting the default PDF reader?

**Comment by dnwk** [comment]
Yes. But I have disabled the PDF reader now. It just gives me a link to download.

**Comment by Yetangitu** [comment]
I'll look into fixing this for the next release, one of these... weeks? I need to fix problems with OPDS Catalog first...

**Comment by Yetangitu** [comment]
Correction, the problem is actually very simple to fix. If you have direct access to your instance you can apply the following patch to get it to work:

```
diff --git a/files_reader/js/plugin.js b/files_reader/js/plugin.js
index 5c4bc3e..5655173 100644
--- a/files_reader/js/plugin.js
+++ b/files_reader/js/plugin.js
@@ -148,7 +148,11 @@ OC.Plugins.register('OCA.Files.FileList', OCA.Files_Reader.Plugin);
 
 // FIXME: Hack for single public file view since it is not attached to the fileslist
 $(document).ready(function(){
-       if ($('#isPublic').val() && ($('#mimetype').val() === 'application/epub+zip'|| $('#mimetype').val() === 'application/x-cbr)')) {
+       if ($('#isPublic').val()
+        && ($('#mimetype').val() === 'application/epub+zip'
+            || $('#mimetype').val() === 'application/pdf'
+            || $('#mimetype').val() === 'application/x-cbr')
+        ) {
                var sharingToken = $('#sharingToken').val();
                var downloadUrl = OC.generateUrl('/s/{token}/download', {token: sharingToken});
                var viewer = OCA.Files_Reader.Plugin;
diff --git a/files_reader/lib/Controller/PageController.php b/files_reader/lib/Controller/PageController.php
index 3306292..14b2224 100644
--- a/files_reader/lib/Controller/PageController.php
+++ b/files_reader/lib/Controller/PageController.php
@@ -135,7 +135,7 @@ class PageController extends Controller {
      */ 
     private function getFileInfo($path) {
         $count = 0;
-        $shareToken = preg_replace("/(?:\/index\.php)?\/s\/([A-Za-z0-9]{15})\/download.*/", "$1", $path, 1,$count);
+        $shareToken = preg_replace("/(?:\/index\.php)?\/s\/([A-Za-z0-9]{15,32})\/download.*/", "$1", $path, 1,$count);
 
         if ($count === 1) {
 
```

This patch makes the preview work for shared files. There is still something left to fix but it is not critical as all it causes is some log spam related to position saving due to the fact that there is no logged in user when reading a shared file.

**Comment by Yetangitu** [comment]
BTW, this will still be included in the next release so if you don't feel like patching manually you can just wait a few days for it to appear.

**Comment by dnwk** [comment]
Thanks

**Comment by Yetangitu** [comment]
Fixed in v1.2.3
